File: restaurant-server/src/main/java/com/cm/restaurant_server/business/domain/entity/Factura.java
package com.cm.restaurant_server.business.domain.entity;

import com.cm.restaurant_server.business.domain.enumeration.EstadoFactura;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import org.hibernate.annotations.SQLRestriction;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Factura extends Base {
    private Long numeroFactura;
    private LocalDate fechaFactura;
    private double totalPagado;
    @Enumerated(EnumType.STRING)
    private EstadoFactura estado;
    @ManyToOne
    private FormaDePago formaDePago;
    @ManyToOne
    private Promocion promocion;
    @OneToMany(cascade = CascadeType.ALL, mappedBy = "factura", orphanRemoval = true)
    @SQLRestriction("eliminado = false")
    List<DetalleFactura> detalleFacturas = new ArrayList<>();
}

File: dashboard/src/types/factura.ts
import type { FormaDePago } from "./formaDePago";
import type { Promocion } from "./promocion";

export enum EstadoFactura {
    PAGADA = "PAGADA",
    ANULADA = "ANULADA",
    SIN_DEFINIR = "SIN_DEFINIR",
}

export interface Factura {
    id: string;
    numeroFactura: number;
    fechaFactura: string; 
    totalPagado: number;
    estado: EstadoFactura;
    formaDePago: FormaDePago | null;
    promocion: Promocion | null;
}

// detalleComanda se ignora por ahora
export interface DetalleFactura {
    id: string;
    cantidad: number;
    subtotal: number;
    factura: Factura | null;
}
